src/AppBundle/Service/TeamMembershipService.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace AppBundle\Service;
+
+use AppBundle\Event\TeamMembershipEvent;
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class TeamMembershipService
+{
+    private $em;
+    private $dispatcher;
+
+    public function __construct(EntityManager $em, EventDispatcherInterface $dispatcher)
+    {
+        $this->em = $em;
+        $this->dispatcher = $dispatcher;
+    }
+
+    public function updateTeamMemberships()
+    {
+        $teamMemberships = $this->em->getRepository('AppBundle:TeamMembership')->findBy(array('isSuspended' => false));
+        foreach ($teamMemberships as $teamMembership) {
+            $endSemester = $teamMembership->getEndSemester();
+            if ($endSemester) {
+                if ($endSemester->getSemesterEndDate() <= $this->em->getRepository('AppBundle:Semester')->findCurrentSemester()->getSemesterStartDate()) {
```

Comment:
Ytelse: Hvis vi henter ut `$currentSemester` på utsiden av loopen så slipper vi å gjøre en query mot databasen for hver `$teamMembership`